### ISSUE TITLE
wxGridCellFloatEditor compile warning on ARM architectures

### ISSUE THREAD

**Issue by swt2c** [wxWidgets/wxWidgets#23101]
### Description

I'm seeing a compile warning when building wx 3.2.1 on Debian Unstable, but only on ARM architectures (arm64, armhf, armel).

```
./src/generic/grideditors.cpp: In member function ‘virtual void wxGridCellFloatEditor::StartingKey(wxKeyEvent&)’:
./src/generic/grideditors.cpp:1052:21: warning: comparison of integer expressions of different signedness: ‘int’ and ‘wxChar’ {aka ‘wchar_t’} [-Wsign-compare]
 1052 |          || keycode == wxNumberFormatter::GetDecimalSeparator() )
      |             ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./src/generic/grideditors.cpp: In member function ‘virtual bool wxGridCellFloatEditor::IsAcceptedKey(wxKeyEvent&)’:
./src/generic/grideditors.cpp:1189:33: warning: comparison of integer expressions of different signedness: ‘const int’ and ‘wxChar’ {aka ‘wchar_t’} [-Wsign-compare]
 1189 |                         keycode == wxNumberFormatter::GetDecimalSeparator() ||
      |                         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
The full build log is available here: https://buildd.debian.org/status/fetch.php?pkg=wxwidgets3.2&arch=arm64&ver=3.2.1%2Bdfsg-3&stamp=1672160163&raw=0

### Platform and version information

- wxWidgets version  you are building: 3.2.1
- wxWidgets port  you are building: wxGTK
- OS  and its version: Debian Unstable
- Compiler  being used: gcc 12.2.0

**Comment by vadz** [comment]
I wonder why don't we get this warning under the other platforms, surely `wchar_t` should be unsigned under them too, and not just under ARM?

Anyhow, the proper fix is, I think, to use `event.GetUnicodeKey()` here as #23103 does. Can you easily check if this avoids the problem, i.e. can the ARM build be run for that PR (as we don't have ARM CI ourselves)?

But for 3.2 we're going to have to just suppress the warning using a cast of `keycode` to `wxChar` as we still support non-Unicode builds there and so can't use `GetUnicodeKey()` unconditionally. Again, if you can easily test this fix, it would be great, otherwise I'll try to set things up for compiling wx for ARM myself.

**Comment by swt2c** [comment]
Sure, I can check it.

**Comment by swt2c** [comment]
Also, sure enough, `wchar_t` is signed on most platforms, but on ARM (32-bit and 64-bit) it is unsigned.  It appears the macro `__WCHAR_UNSIGNED__` will be defined on (gcc) platforms where ARM is unsigned.

**Comment by swt2c** [comment]
Yes, I can confirm #23103 takes care of the warnings on ARM64.

**Comment by vadz** [comment]
Thanks for testing! I've pushed this now and will push the simpler fix to 3.2 slightly later.

I'm also adding an ARM CI build in #23104 to avoid such problems in the future.

**Comment by swt2c** [comment]
Thanks for the unexpectedly quick fix on this and Happy New Year!
